_pages/dev/ssr/server-side-rendering-optimization.md

```diff
@@ -0,0 +1,233 @@
+
+  By default, `renderKeyResolver` uses `req.originalUrl`.
+
+- `renderingStrategyResolver` is a function that accepts `(req: Request) => RenderingStrategy`, which allows you to define custom rendering strategies for each request. The available `RenderingStrategy` strategies work as follows:
+
+  - `ALWAYS_CSR` always returns client-side rendered pages
+  - `DEFAULT` is the default behavior
+  - `ALWAYS_SSR` aLways returns server-side rendered pages
+
+- `forcedSsrTimeout` is a number that indicates the time (in milliseconds) to wait for rendered pages when the render strategy for the request is set to `ALWAYS_SSR`. This prevents SSR rendering from blocking resources for too long if the server is under heavy load, or if the page contains errors.
+
+  The default value is `60000` milliseconds (that is, 60 seconds).
+
+- `debug` is a boolean that, when set to `true`, enables extra logs that are useful for troubleshooting SSR issues. In production environments, you should set `debug` to `false` to avoid an excessive number of logs. Regardless, the SSR timeout log will capture `SSR rendering exceeded timeout...` even if the `debug` flag is set to `false`.
+
+  **Note**: This property is available in version 3.1.0 and later.
+
+  The default value is `false`.
+
+## Troubleshooting
+
+The following sections describe how to resolve potential issues with SSR.
+
+### Verifying That Your Storefront is Running in SSR Mode
+
+If you are encountering issues with SSR, a good first step is to verify that your storefront is running in SSR mode. The following sections describe how to check if your storefront is rendering SSR pages correctly.
+
+#### Using the Curl Command
+
+On Linux and MacOs, you can use the `curl` command to check if your storefront is running in SSR mode. Run the following command, replacing `YOUR_SITE_URL` with the URL of the site you want to test:
+
+```sh
+curl YOUR_SITE_URL
+```
+
+This should return an HTML markup response that contains rendered elements, such as the following:
+
+```html
+  [...]
+  -root _nghost-sc293="" ng-version="10.1.6"><cx-storefront _ngcontent-sc293="" tabindex="0" s="LandingPage2Template    stop-navigating"><cx-skip-link><div tabindex="-1" class=""><button> Skip to Header </on><button> Skip to Main Content </  button><button> Skip to Footer </button><!----></div><!----></cx-skip-link><header iplink="cx-header" class=""     tabindex="-1"><cx-page-layout section="header" class="header"><cx-page-slot position="PreHeader" class="PreHeader     has-components"><cx-hamburger-menu><button type="button" aria-label="Menu" -controls="header-account-container,     header-categories-container, header-locale-container" class="cx-hamburger" -expanded="false"><span    class="hamburger-box"><span class="hamburger-inner"></span></span></button></hamburger-menu>
```

Comment:
`-root _nghost-sc293...` should be `<app-root_nghost-sc293...`

Reply:
Fixed 🙂